src/backend/expungeservice/endpoints/users.py

```diff
@@ -110,6 +110,62 @@ def post(self):
         return jsonify(response_data), 201
 
 
+    @user_auth_required
+    def put(self, user_id):
+        """
+        Update the user entry with new values for one or more of the user data fields:
+        email, name, group_name, password, or admin
+        """
+        user_db_data = user.read(g.database, user_id)
+        if not user_db_data:
+            error(404, "User id not recognized")
+
+        if not g.logged_in_user_is_admin and g.logged_in_user_id != user_id:
+            error(403, "Logged in user not admin and doesn't match requested user id.")
+
+
+        data = request.get_json()
+
+        if data is None:
+            error(400, "No json data in request body")
+
+        if not any([key in data.keys() for key in ["email", "name", "group_name",
+                                 "password", "admin"]]):
```

Comment:
Correct me if I'm wrong, but user_auth_required checks if there is a currently logged in user. This would allow for the user to upgrade themselves to an admin?

Reply:
Wow, yep that is an excellent point and needs to be changed.